Intervalos/src/LM/intervalos.jl

```diff
@@ -0,0 +1,293 @@
+            supremo=max(nextfloat(in_a*in_b),nextfloat(in_a*su_b),nextfloat(su_a*in_b),nextfloat(su_a*su_b))
+        end
+        return Intervalo(infimo,supremo)
+    end
+end
+
+*(a::Intervalo,b::Real)=*(b,a)
+
+import Base: /
+
+function /(a,b::Intervalo)
+    in_b=getfield(b, :infimo)
+    su_b=getfield(b, :supremo)
+    
+    if 0 ∈ b
+        return Intervalo(-Inf, Inf)    
+    elseif typeof(a)<:Real
+        infimo=min(prevfloat(a/in_b),prevfloat(a/su_b))
+        supremo=max(nextfloat(a/in_b),nextfloat(a/su_b))
+        return Intervalo(infimo,supremo)
+    else
+        in_a=getfield(a, :infimo)
+        su_a=getfield(a, :supremo)
+        
+        infimo=min(prevfloat(in_a/in_b),prevfloat(in_a/su_b),prevfloat(su_a/in_b),prevfloat(su_a/su_b))
+        supremo=max(nextfloat(in_a/in_b),nextfloat(in_a/su_b),nextfloat(su_a/in_b),nextfloat(su_a/su_b))
+        return Intervalo(infimo,supremo)
+    end
+end
+
+# Definimos la función para las potencias enteras
+
+import Base: ^
+
+function ^(a::Intervalo,b::Int64)
+    if b==0
+        return Intervalo(1,1)
+    end
+    
+    if isnan(getfield(a, :infimo))
+        return a
+    end
+    
+    a_in=prevfloat(getfield(a,:infimo)^b)
+    a_su=nextfloat(getfield(a,:supremo)^b)
+    
+    if b%2!=0
+        return Intervalo(a_in,a_su)
+    else
+        if getfield(a,:infimo) ≤ 0 && getfield(a,:supremo) ≥ 0
+            supremo= max(a_in,a_su)
+            return Intervalo(0,supremo)
+        else 
+            infimo=min(a_in,a_su)
+            supremo=max(a_in,a_su)
+            return (infimo,supremo)
+        end
+    end
+end
+
+#Definimos la función division_extendida de acuerdo a lo visto en clase.
+
+function division_extendida(a::Intervalo,b::Intervalo)
+    in_a=getfield(a, :infimo)
+    su_a=getfield(a, :supremo)
+    in_b=getfield(b, :infimo)
+    su_b=getfield(b, :supremo)
+    
+    if 0 ∉ b
+        return a/b
+    elseif 0 ∈ a && 0 ∈ b
+        return Intervalo(-Inf,Inf)
+    elseif su_a <0 && in_b < su_b==0
+        return Intervalo(prevfloat(su_a/in_b),Inf)
+    elseif su_a <0 && in_b < 0 < su_b
+        println(Intervalo(su_a/in_b,Inf)," ∪ ",Intervalo(-Inf,su_a/su_b))
+        return (Intervalo(prevfloat(su_a/in_b),Inf),Intervalo(-Inf,nextfloat(su_a/su_b)))
+    elseif su_a <0 && 0==in_b < su_b
+        return Intervalo(-Inf,nextfloat(su_a/su_b))
+    elseif 0 < in_a && in_b < su_b==0
+        return Intervalo(-Inf,nextfloat(in_a/in_b))
+    elseif 0 < in_a && in_b < 0 < su_b
+        println(Intervalo(in_a/su_b,Inf)," ∪ ",Intervalo(-Inf,in_a/in_b))
+        return (Intervalo(prevfloat(in_a/su_b),Inf), Intervalo(-Inf,nextfloat(in_a/in_b)))
+    elseif 0 < in_a && 0==in_b < su_b
+        return Intervalo(prevfloat(in_a/su_b),Inf)
+    elseif 0 ∉ a && b==Intervalo(0,0)
+        return intervalo_vacio(typeof(in_a))
+    end
+end
+
+/(a::Intervalo,b::Intervalo)=division_extendida(a,b)
+
+import Base: inv
+
+inv(a::Intervalo)=division_extendida(Intervalo(1),a)
```

Comment:
```suggestion
inv(a::Intervalo)=1/a
```
Cambia esto por la división normal